packages/expo-media-library/ios/MediaLibraryModule.swift

```diff
@@ -339,6 +339,60 @@ public class MediaLibraryModule: Module, PhotoLibraryObserverHandler {
     }
   }
 
+  private func handleLivePhoto(asset: PHAsset, shouldDownloadFromNetwork: Bool, result: [String: Any?], promise: Promise) {
+    let livePhotoOptions = PHLivePhotoRequestOptions()
+    livePhotoOptions.isNetworkAccessAllowed = shouldDownloadFromNetwork
+    var updatedResult = result
+      updatedResult["pairedVideoAsset"] = nil
+
+    PHImageManager.default()
+      .requestLivePhoto(for: asset, targetSize: PHImageManagerMaximumSize, contentMode: .aspectFit, options: livePhotoOptions) { livePhoto, _ in
+      guard let livePhoto = livePhoto,
+        let videoResource = PHAssetResource.assetResources(for: livePhoto)
+        .first(where: { $0.type == .pairedVideo }) else {
+        promise.resolve(updatedResult)
+        return
+      }
+      self.writePairedVideoAsset(videoResource: videoResource, asset: asset, result: updatedResult, promise: promise)
+      }
+  }
+
+  private func writePairedVideoAsset(videoResource: PHAssetResource, asset: PHAsset, result: [String: Any?], promise: Promise) {
+    let fileName = videoResource.originalFilename
+    let tempDir = FileManager.default.temporaryDirectory
+    let fileExt = getFileExtension(from: fileName).replacingOccurrences(of: ".", with: "")
+    let tempId = UUID().uuidString
+    let fileUrl = tempDir.appendingPathComponent(tempId).appendingPathExtension(fileExt)
+    PHAssetResourceManager.default().writeData(for: videoResource, toFile: fileUrl, options: nil) { error in
+      guard error == nil else {
+        promise.resolve(result)
+        return
+      }
+      let avAsset = AVAsset(url: fileUrl)
+      let duration = avAsset.duration.seconds
+      var pairedVideoAsset: [String: Any?] = [
```

Comment:
`[String: Any?]` - saw how that was how it was done in e.g. `exportAsset`, so used the same idea here as well.